lib/Doctrine/ORM/EntityManager.php

```diff
@@ -157,8 +156,12 @@ protected function __construct(Connection $conn, Configuration $config, EventMan
         $this->metadataFactory->setCacheDriver($this->config->getMetadataCacheImpl());
 
         $this->repositoryFactory = $config->getRepositoryFactory();
-        $this->unitOfWork        = new UnitOfWork($this);
-        $this->proxyFactory      = new ProxyFactory(
+
+        $unitOfWorkClassName  = $config->getUnitOfWorkClassName();
+        $this->unitOfWork     = new $unitOfWorkClassName($this);
```

Comment:
@stof not sure I understand where we have sensible logic in UnitOfWork. There are cases where you need to customize UnitOfWork, however having no interfaces force to copy-paste EntityManager, UnitOfWork, ProxyFactory, collections, persisters, hydrators etc.